### PR TITLE
Add metadata for pip / PyPI.org

### PR DESCRIPTION
**Pull request by schiotz**
Add metadata for uploading to pypi.org so it becomes pip-installable.

~Work-in-progress: setup.py needs to be modified to work with Windows as well.~

### CODE REVIEW

**Comment by schiotz** [comment]
Hi @jacobjma 

I think this is ready for merging now.  There was no way I could get PyQSTEM to install on Windows just using setup.py.  It looks like some kind of binary incompatibility between something (compilers, Anaconda, the pre-build FFTW, I could not work it out).

Using @msarahan's conda recipe, I finally got it to work on Windoze.

I think it should be ready to merge.  To make PyQSTEM really easy to install, it should be uploaded to PyPI and conda forge.  I suggest the following steps after merging this:

* Generate a source distribution with `python setup.py sdist` and upload it with `twine`. 
* Update conda.recipe/meta.yaml to contain the PyPI url.  Update the installation instructions.
* Upload the conda recipe.

I think it is best that you do the first one, but I am willing to do the next two (admittedly, I have never uploaded anything to Anaconda, but how hard can it be ...  well after the awful build experience I am slightly worried...)

**Comment by msarahan** [comment]
Take your recipe from here over to https://github.com/conda-forge/staged-recipes - you'll need to tweak it a bit, but not too much.

**Comment by schiotz** [comment]
Thanks, @msarahan 

I would never have got it to work without your recipe file!  

Even with it, it is a nightmare of compiler versions and variables.  The Anaconda prompt cannot see the compiler, and the Visual Studio prompt cannot see Anaconda, unless it is setting the path when being installed, which they warn strongly against.  Probably, I am doing something else wrong, but I am certainly **not** a Windows developer :-)   At least starting with a sensible conda recipe made it doable.

File: conda.recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  path: ..
+  # git_url: https://github.com/jacobjma/PyQSTEM
+  # git_tag:  (commit id or tag of release to build, defaults to master)
+
+build:
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - cython
```
Comment (msarahan) [inline conda.recipe/meta.yaml]: This makes the compilers easier to deal with.  You can read more at https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html

```suggestion
    - {{ compiler("c") }}
    - {{ compiler("cxx") }}
    - cython
```

File: setup.py [hunk 2]
```diff
@@ -38,16 +51,26 @@
...
+      license='GPLv3',
+      data_files = [("", ["License.txt"])],
+      platforms = ["Windows", "Linux", "Mac OS-X", "Unix"],
+      install_requires=['matplotlib!=3.0.0', 'ase', 'Cython', 'scikit-image', 'pillow'],
+      ext_modules=cythonize(Extension('pyqstem.qstem_interface',
+                                      sources=sources,
+                                      library_dirs=library_dirs,
+                                      libraries=libraries,
+                                      include_dirs=include_dirs,
+                                      extra_compile_args=['-std=c++11','-D MS_WIN64'],
```
Comment (msarahan) [inline setup.py]: Is this -D MS_WIN64 going to do anything silly if it runs on linux/mac?

Comment (schiotz) [reply setup.py]: Good catch!  I don't know if it does any harm, but it is certainly wrong.  Also, ``-std=c++11`` cause a warning with the Windows compilers, and does not seem necessary there.

File: conda.recipe/meta.yaml (hunk 1 above)
Comment (schiotz) [reply conda.recipe/meta.yaml]: This makes it possible to compile in the normal Anaconda command prompt window.  Unfortunately, it causes linkage to fail (both in Anaconda and Visual Studio windows) as it fails to find fftw3.lib.  I am confused.

**Comment by schiotz** [comment]
Hi Jacob @jacobjma 

I can confirm that this also works on Anaconda for Mac, and I think it is ready to be merge.

/Jakob

**Comment by jacobjma** [comment]
Hi Jakob @schiotz 

Thanks for doing this. I will look at uploading it to PyPI tomorrow.

**Pull request merged**